### PR TITLE
tests: coverage tests

### PR DESCRIPTION
**Pull request by Clintonrocha98**
This pull request introduces significant improvements to the application transitions feature, focusing on enhanced test coverage, improved factory methods for test setup, and some code cleanup. The main changes include comprehensive feature tests for application status transitions, new factory helpers to streamline test scenarios, and the removal of an unused policy method.

**Test Coverage Improvements:**

* Added a full-featured test suite for the abstract application transition logic, covering stage history creation, event dispatching, rollback behavior, and metadata handling (`tests/Feature/Transitions/AbstractTransitionTest.php`).
* Added dedicated tests for `InReviewTransition` and `InProgressTransition`, ensuring all edge cases and expected behaviors (including error scenarios) are covered for these transitions (`tests/Feature/Transitions/InReviewTransitionTest.php`, `tests/Feature/Transitions/InProgressTransitionTest.php`). [[1]](diffhunk://#diff-2a8a5cd6d1d082febeb4469d327efc1bdeae47b34c0f38f8424125a6495e10c5R1-R145) [[2]](diffhunk://#diff-d6d4e65a4a4f747452bfd890d61d7a206515007df9d2644172240614d59b471cR1-R182)

**Test Factory Enhancements:**

* Introduced new helper methods to `ApplicationFactory` for easier and more deterministic setup of application and stage scenarios in tests, including `withStatus`, `withIsolatedStages`, `withNoStages`, and `createWithStage`. These helpers allow tests to precisely control application status and stage relationships (`database/factories/ApplicationFactory.php`).

**Code Cleanup:**

* Removed the unused `transition` method from `ApplicationPolicy`, along with its related import, to simplify the policy and avoid confusion (`src/Policies/ApplicationPolicy.php`). [[1]](diffhunk://#diff-6f1a87fa6dcdea1922e03638f1b392872f8934361efa95e9d6b7e87ae47eac33L51-L71) [[2]](diffhunk://#diff-6f1a87fa6dcdea1922e03638f1b392872f8934361efa95e9d6b7e87ae47eac33L7)

These changes collectively make the application's transition logic more robust and maintainable, with much better test coverage and easier test setup.

## Release Notes

* **New Features**
  * Enhanced application transition workflow with improved status validation and stage management controls.
  * Updated team member invitation process to guide users to set their own password via password recovery.

* **Documentation**
  * Updated team invitation email instructions with clearer password setup guidance.

* **Chores**
  * Updated Filament, Laravel, and supporting dependencies.
  * Refactored test infrastructure for improved consistency.

### CODE REVIEW

**Comment by Clintonrocha98** [comment]
## 🔴 Bug: State machine travada em `OfferAcceptedTransition` e `OfferExtendedTransition`

Ambas as transitions tinham um bug crítico em `processStep()`: ignoravam `$data->toStatus` e hardcodavam seu próprio status — tornando `Hired` e `OfferAccepted` inacessíveis pela state machine.

---

### `OfferAcceptedTransition`

`choices()` permite somente `Hired`, mas `processStep()` sempre persistia `OfferAccepted`. Como resultado, `$fromStatus === $toStatus` no `handle()` — o evento `ApplicationStatusChanged` nunca era despachado e o `HiredTransition` se tornava inacessível.

```php
// Antes — status nunca avançava para Hired
public function processStep(TransitionData $data): void
{
    $this->application->update([
        'status' => ApplicationStatusEnum::OfferAccepted, // hardcoded, ignora $data->toStatus
    ]);
}

// Depois
public function processStep(TransitionData $data): void
{
    $this->application->update([
        'status' => $data->toStatus,
    ]);
}
```

---

### `OfferExtendedTransition`

`choices()` permite `OfferAccepted`, `OfferDeclined` e `Withdrawn`, mas `processStep()` sempre persistia `OfferExtended` — nenhuma transição de saída funcionava.

Além da correção do status, os campos de oferta (`offer_amount`, `offer_extended_at`, etc.) passaram a ser atualizados **somente** quando `toStatus === OfferAccepted`, já que não fazem sentido ao declinar ou retirar a oferta.

```php
// Antes — status nunca saía de OfferExtended
public function processStep(TransitionData $data): void
{
    $this->application->update([
        'status' => ApplicationStatusEnum::OfferExtended, // hardcoded, ignora $data->toStatus
        'offer_extended_at' => $data->offerExtendedAt ?? now(),
        'offer_extended_by' => $data->byUserId,
        'offer_amount' => $data->offerAmount ?? $this->application->offer_amount,
        'offer_response_deadline' => $data->offerResponseDeadline ?? $this->application->offer_response_deadline,
    ]);
}

// Depois
public function processStep(TransitionData $data): void
{
    $payload = ['status' => $data->toStatus];

    if ($data->toStatus === ApplicationStatusEnum::OfferAccepted) {
        $payload += [
            'offer_extended_at' => $data->offerExtendedAt ?? now(),
            'offer_extended_by' => $data->byUserId,
            'offer_amount' => $data->offerAmount ?? $this->application->offer_amount,
            'offer_response_deadline' => $data->offerResponseDeadline ?? $this->application->offer_response_deadline,
        ];
    }

    $this->application->update($payload);
}
```

---

Os testes foram corrigidos para refletir o comportamento correto — anteriormente documentavam o comportamento quebrado como "intencional".

**Comment by Clintonrocha98** [comment]
## 🟠 Nova migration: unique index em `teams.slug`

A coluna `slug` da tabela `teams` não possuía constraint de unicidade no banco — apenas validação na camada de aplicação via Filament. Isso abre brecha para race condition em cadastros simultâneos, onde dois requests podem passar pela validação antes de qualquer um inserir no banco.

Foi criada a migration `add_unique_index_to_slug_on_teams_table` adicionando `UNIQUE` na coluna.

```php
public function up(): void
{
    Schema::table('teams', function (Blueprint $table): void {
        $table->unique('slug');
    });
}
```

✅ **Verificado em produção**: nenhum team com slug duplicado encontrado — a migration pode ser aplicada sem risco.

File: app-modules/term/tests/Feature/TermResourceTest.php [hunk 1]
```diff
@@ -15,11 +14,9 @@
 use function Pest\Livewire\livewire;
 
 beforeEach(function (): void {
-    Filament::setCurrentPanel(Filament::getPanel('admin'));
+    filament()->setCurrentPanel(FilamentPanel::Admin->value);
```
[the remark this answers is not in the document]
Comment (Clintonrocha98) [reply app-modules/term/tests/Feature/TermResourceTest.php:17]: I'm using filament v5 and the setCurrentPanel method accepts string.
